ios/Breez Notification Service Extension/BreezManager.swift

```diff
@@ -11,55 +11,66 @@ fileprivate var log = Logger(
+            if BreezManager.breezSDK == nil {
+                BreezManager.breezSDK = try BreezManager.connectSDK()
+            }
+            return BreezManager.breezSDK
+        }
+    }
     
-    init(paymentListener: @escaping PaymentListener) {
-        self.paymentListener = paymentListener
+    static func unregister() {
+        BreezManager.queue.sync { [] in
+            BreezManager.sdkListener = nil            
+        }
     }
     
-    func onEvent(e: BreezEvent) {
-        switch e {
-        case .invoicePaid(details: let details):
-            log.info("Received payment. Bolt11: \(details.bolt11)\nPayment Hash:\(details.paymentHash)")
-            if details.payment != nil {
-                paymentListener(details.payment!)
-            }
-            return
-        default:
-            break
+    static func connectSDK() throws -> BlockingBreezServices? {
+        log.trace("connectSDK()")
+        try setLogStream(logStream: SDKLogListener(logger: NotificationService.logger))
+        
+        // Create the default config
+        let apiKey = try Environment.glApiKey()
+        log.trace("API_KEY: .\(apiKey)")
+        var config = defaultConfig(envType: EnvironmentType.production, apiKey: apiKey,
+                                   nodeConfig: NodeConfig.greenlight(
+                                    config: GreenlightNodeConfig(partnerCredentials: nil, inviteCode: nil)))
+        
+        config.workingDir = FileManager
+            .default.containerURL(forSecurityApplicationGroupIdentifier: "group.F7R2LZH3W5.com.cBreez.client")!
+            .absoluteString
+        
+        // Construct the seed
+        let mnemonic = CredentialsManager.shared.restoreMnemonic() ?? ""
+        log.trace("mnemonic: .\(mnemonic)")
+        let seed = try? mnemonicToSeed(phrase: mnemonic)
+        // Connect to the Breez SDK make it ready for use
+        guard seed != nil else {
+            throw SdkError.Generic(message: "seed not found")
         }
+        log.trace("Connecting to Breez SDK")
+        let breezSDK = try connect(config: config, seed: seed!, listener: BreezManagerListener())
+        log.trace("Connected to Breez SDK")
+        return breezSDK
     }
 }
 
-func connectSDK(paymentListener: @escaping PaymentListener) throws -> BlockingBreezServices? {
-    log.trace("connectSDK()")
+class BreezManagerListener: EventListener {
     
-    // Create the default config
-    let apiKey = try Environment.glApiKey()
-    log.trace("API_KEY: .\(apiKey)")
-    var config = defaultConfig(envType: EnvironmentType.production, apiKey: apiKey,
-                               nodeConfig: NodeConfig.greenlight(
-                                config: GreenlightNodeConfig(partnerCredentials: nil, inviteCode: nil)))
+    func onEvent(e: BreezEvent) {
+        BreezManager.queue.async { [] in
+            if let l = BreezManager.sdkListener {
+                l.onEvent(e: e)
+            }
```

Comment:
Can we use optional chaining here?
```suggestion
            BreezManager.sdkListener?.onEvent(e: e)
```

Reply:
Done